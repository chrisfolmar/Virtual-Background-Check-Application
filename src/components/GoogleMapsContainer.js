import React from 'react';
import { GoogleApiWrapper, InfoWindow, Map, Marker } from 'google-maps-react';

class GoogleMapsContainer extends React.Component {
  render() {
    const mapStyle = {
      width: '100%', 
      height: '30vh'
    }

    return (
        <Map
          item
          style = {mapStyle}
          xs = { 12 }
          google = { this.props.google }
          zoom = { this.props.zoom }
          center = { this.props.center }
          initialCenter = { this.props.center } 
          mapTypeControl = { false }
          streetViewControl = { false }
        >
          <Marker
            position = {this.props.center}
          />
        </Map>

    );
  }
}
export default GoogleApiWrapper({
    api: (process.env.REACT_APP_GOOGLE_KEY)
})(GoogleMapsContainer)