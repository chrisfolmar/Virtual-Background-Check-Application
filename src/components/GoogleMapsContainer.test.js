import React from 'react';
import ReactDOM from 'react-dom';
import GoogleMapsContainer from './GoogleMapsContainer';
import { shallow, mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json'

const center = {
	lat: 25,
	lng: -72
}

describe('GoogleMapsContainer', () => {

	it('renders fully without crashing', () => {
		const component = mount(<GoogleMapsContainer center={center} zoom={10} />);
		component.unmount();
	});

	it('full snapshot rendering', () => {
		const component = mount(
			<GoogleMapsContainer />
			);
		expect(mountToJson(component)).toMatchSnapshot();
		component.unmount();
	});
});