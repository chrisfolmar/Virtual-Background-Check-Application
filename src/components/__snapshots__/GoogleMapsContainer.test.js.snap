// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMapsContainer full snapshot rendering 1`] = `
<Wrapper>
  <DefaultLoadingContainer>
    <div>
      Loading...
    </div>
  </DefaultLoadingContainer>
</Wrapper>
`;
